src/components/Auth.js

```diff
@@ -0,0 +1,207 @@
+import {
+  Box, Button, CircularProgress, Typography,
+} from '@material-ui/core';
+import fetchIntercept from 'fetch-intercept';
+import * as jwt from 'jsonwebtoken';
+import Keycloak from 'keycloak-js';
+import { PropTypes } from 'prop-types';
+import React, {
+  createContext, useContext, useEffect,
+  useLayoutEffect, useMemo, useRef,
+} from 'react';
+import { useMutation } from 'react-query';
+import { Route } from 'react-router-dom';
+
+import api from '@/services/api';
+
+const dbRoles = {
+  admin: 'admin',
+  regular: 'regular',
+  readonly: 'readonly',
```

Comment:
there is also a manager role, tho i don't know if we had this yet in the previous version

Reply:
this was just copied from src/services/auth.js